### ISSUE TITLE
tsconfig should exclude files from public directory

### ISSUE THREAD

**Issue by ianschmitz** [wmonk/create-react-app-typescript#159]
### Is this a bug report?

Yes

### Can you also reproduce the problem with npm 4.x?

Yes

### Environment

1. `npm ls react-scripts-ts` (if you haven’t ejected): 2.6.0
2. `node -v`: 8.4.0
3. `npm -v`: 5.3.0

Then, specify:

1. Operating system: Windows 10

### Steps to Reproduce

(Write your steps here:)

1.  Copy `node_modules` folder to `public` directory
2. Run `npm start`
3. Notice errors during startup in console. For example i see error: 
```
C:\Geocortex\geocortex-app-designer\node_modules\@types\node\index.d.ts
(43,11): error TS2451: Cannot redeclare block-scoped variable 'Error'.
```

### Expected Behavior

Only files from `src/` should be included in build.

### Actual Behavior

Files from `public/` are included.

An easy solution was for me to either:

1. Add `public` to `exclude` array in `tsconfig.json`. or
1. Repalce `exclude` array with `include` array with `src/**/*` as only element.

It seems to me that 2 is preferable if we intend only to be looking at the `src` directory. There may be drawbacks that I'm not aware of.

**Comment by ianschmitz** [comment]
After considering #152, a potential solution could be something like:
```JSON
  "include": [
    "src/**/*"
  ],
  "exclude": [
    "src/setupTests.ts",
    "src/**/*.test.{ts,tsx}",
    "src/**/*.spec.{ts,tsx}",
    "src/**/__tests__/*"
  ]
```

**Comment by DorianGrey** [comment]
While adding an exclusion or an explicit inclusion should not be a problem, I don't understand why anyone should do this:
> Copy node_modules folder to public directory

Can you elaborate the use case for this? It's just that this folder is not intended for source files, neither your own nor those from libraries. If you really have to serve content from the `node_modules` directory directly - and you should not need to do this in almost every case - I suppose there are better ways to achieve this.

**Comment by wmonk** [comment]
Yes agree with @DorianGrey, seems like an odd use case. However, I do think we should only be including the `src` directory in tsconfig, as this fits with how webpack is setup to only allow the importing of files within this dir. Happy to merge a PR for this 😄

**Comment by ianschmitz** [comment]
Sorry I should have elaborated. Im not proposing to try to support node modules in the public directory. It was simply an easy way to reproduce the fact that the it appears files inside public are being consumed by typescript.
